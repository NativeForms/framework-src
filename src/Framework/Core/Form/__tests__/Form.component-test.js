import React from 'react';
import { shallow } from 'enzyme';
import FormComponent from '../Form.component';

//schema
import schemaControlButton from '../Control/Button/Button.schema';
import schemaControlLink from '../Control/Link/Link.schema';
import schemaFieldTextInput from '../Field/TextInput/TextInput.schema';
//message
import messageControlButton from '../Control/Button/Button.i18n';
import messageControlLink from '../Control/Link/Link.i18n';
import messageFieldTextInout from '../Field/TextInput/TextInput.i18n';

const propsArray = [
  {
    schema: schemaControlButton,
    message: messageControlButton.en
  },
  {
    schema: schemaControlLink,
    message: messageControlLink.en
  },
  {
    schema: schemaFieldTextInput,
    message: messageFieldTextInout.en
  }
];

describe('components <FormComponent />', () => {
  let wrapper;
  propsArray.forEach((props) => {
    it('renders form components correctly', () => {
      wrapper = shallow(
        <FormComponent {...props} />
    );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
