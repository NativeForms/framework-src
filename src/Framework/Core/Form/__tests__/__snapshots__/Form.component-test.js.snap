// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components <FormComponent /> renders form components correctly 1`] = `
<IntlProvider
  messages={Object {}}
>
  <Styled(Form)
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <ControlComponent
      attributes={
        Object {
          "label": Object {
            "code": "myLabel",
          },
        }
      }
      type="button"
    />
  </Styled(Form)>
</IntlProvider>
`;

exports[`components <FormComponent /> renders form components correctly 2`] = `
<IntlProvider
  messages={Object {}}
>
  <Styled(Form)
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <ControlComponent
      attributes={
        Object {
          "label": Object {
            "code": "googleLabel",
          },
          "route": "http://www.google.com",
        }
      }
      type="link"
    />
  </Styled(Form)>
</IntlProvider>
`;

exports[`components <FormComponent /> renders form components correctly 3`] = `
<IntlProvider
  messages={Object {}}
>
  <Styled(Form)
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <FieldComponent
      attributes={
        Object {
          "label": "Key",
        }
      }
      type="text"
    />
    <FieldComponent
      attributes={
        Object {
          "hiddenText": true,
          "label": "Secret",
        }
      }
      type="text"
    />
    <FieldComponent
      attributes={
        Object {
          "clearButton": true,
          "label": "Value 1",
          "value": "Delete Me",
        }
      }
      type="text"
    />
    <FieldComponent
      attributes={
        Object {
          "defaultText": "Replace Me",
          "label": "Value 2",
        }
      }
      type="text"
    />
    <FieldComponent
      attributes={
        Object {
          "headerLabel": "Header Label",
        }
      }
      type="text"
    />
    <FieldComponent
      attributes={
        Object {
          "headerLabel": "Description",
          "multiline": true,
          "value": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Itaque hic ipse iam pridem est reiectus; De ingenio eius in his disputationibus, non de moribus quaeritur. Nunc haec primum fortasse audientis servire debemus. Atqui iste locus est, Piso, tibi etiam atque etiam confirmandus, inquam;",
        }
      }
      type="text"
    />
    <FieldComponent
      attributes={
        Object {
          "defaultText": Object {
            "code": "defaultText",
          },
        }
      }
      type="text"
    />
  </Styled(Form)>
</IntlProvider>
`;
